layers/+lang/c-c++/funcs.el

```diff
@@ -182,3 +185,161 @@ and the arguments for flyckeck-clang based on a project-specific text file."
+  (defun c-c++/inheritance-hierarchy-inv () (interactive) (spacemacs//c-c++-lsp-funcall-interactively nil "-inheritance-hierarchy" t))
+  (defun c-c++/member-hierarchy () (interactive) (spacemacs//c-c++-lsp-funcall-interactively-no-args nil "-member-hierarchy"))
+  (defun c-c++/freshen-index () (interactive) (spacemacs//c-c++-lsp-funcall-interactively nil "-freshen-index"))
+  (defun c-c++/preprocess-file () (interactive) (spacemacs//c-c++-lsp-funcall-interactively nil "-preprocess-file")))
+
+(defun spacemacs//c-c++-lsp-bind-keys-for-mode (mode)
+  "Bind LSP backend functions for the specificed mode."
+  (spacemacs/set-leader-keys-for-major-mode mode
+    ;; goto
+    ;; FIXME gf conflicts with `lsp-ui-flycheck-list' binding in lsp layer
+    ;; Better corrected in lsp-layer? rebinding here for now
+    "ge" #'lsp-ui-flycheck-list
+    "gf" 'find-file-at-point
+    "gF" 'ffap-other-window
+    "g&" #'c-c++/references-address
+    "gR" #'c-c++/references-read
+    "gW" #'c-c++/references-write
+    "gc" #'c-c++/callers
+    "gv" #'c-c++/vars
+    ;; help/hierarchy
+    "hb" #'c-c++/base ;;Replace this with lsp-goto-implementation in lsp-layer?
+    "hc" #'c-c++/call-hierarchy
+    "hC" #'c-c++/call-hierarchy-inv
+    "hi" #'c-c++/inheritance-hierarchy
+    "hI" #'c-c++/inheritance-hierarchy-inv
+    "hm" #'c-c++/member-hierarchy
+    ;; lsp/backend
+    "lf" #'c-c++/freshen-index
+    "lp" #'c-c++/preprocess-file))
+
+(defun spacemacs//c-c++-lsp-customise-lsp-ui-peek ()
+  "Wrap some backend-specific features using the `lsp-ui-peek' functions provided by the `lsp' layer (`lsp-ui' package)"
+  (spacemacs//c-c++-lsp-call-function "spacemacs//c-c++-lsp-" "-customise-lsp-ui-peek")
+
+  (defun c-c++/callers () (interactive) (lsp-ui-peek-find-custom 'callers (spacemacs//c-c++-lsp-string "$" "/callers")))
+  (defun c-c++/vars () (interactive) (lsp-ui-peek-find-custom 'vars (spacemacs//c-c++-lsp-string "$" "/vars")))
+
+  (defun c-c++/references-address ()
+    (interactive)
+    (lsp-ui-peek-find-custom
+      'address "textDocument/references"
+      (plist-put (lsp--text-document-position-params) :context
+        '(:role 128))))
+
+  (defun c-c++/references-read ()
+    (interactive)
+    (lsp-ui-peek-find-custom
+      'read "textDocument/references"
+      (plist-put (lsp--text-document-position-params) :context
+        '(:role 8))))
+
+  (defun c-c++/references-write ()
+    (interactive)
+    (lsp-ui-peek-find-custom
+      'write "textDocument/references"
+      (plist-put (lsp--text-document-position-params) :context
+        '(:role 16)))))
+
+(defun spacemacs//c-c++-lsp-cquery-customise-lsp-ui-peek ()
+  (defun c-c++/base () (interactive) (lsp-ui-peek-find-custom 'base "$cquery/base")))
+
+(defun spacemacs//c-c++-lsp-ccls-customise-lsp-ui-peek ()
+  (defun c-c++/base ()
+    (interactive)
+    (lsp-ui-peek-find-custom 'base "$ccls/inheritanceHierarchy"
```

Comment:
`$ccls/inheritanceHierarchy` -> `$ccls/inheritance`

Reply:
Hey Fangrui,
Will make the changes Friday. A gentle suggestion from a lazy integrator... where it doesn't compromise your vision too much, keeping the ccls API close to that of cquery might encourage uptake, as it makes it easier to support both as drop-in alternatives.